components/locale_canonicalizer/src/locale_canonicalizer.rs

```diff
@@ -186,11 +186,23 @@ fn update_langid(
     entry: &LanguageIdentifier,
     langid: &mut LanguageIdentifier,
 ) -> CanonicalizationResult {
+    let language_unmodified = langid.language.clone();
+    let script_unmodified = langid.script.clone();
+    let region_unmodified = langid.region.clone();
```

Comment:
`question`: wouldn't it be much cheaper to register changes, rather than clone and compare?

I mean:

```rust
let language_modified = false;

if (...) {
    language_modified = true;
}

if !language_modified {
    return CanonicalizationResult::Unmodified;
}
```
?

Reply:
Done, please check the PR now